controllers/machinedeletionremediation_controller.go

```diff
@@ -282,23 +279,41 @@ func (r *MachineDeletionRemediationReconciler) getNodeFromMdr(ctx context.Contex
 	return node, nil
 }
 
-// getMachine retrieves a machine from the cluster based on the remediation.
-// It returns the machine, a boolean indicating whether the machine was expected to be found,
-// and an error if any occurred during the retrieval process.
+// getMachine retrieves a Machine from the cluster based on the remediation.
+// It returns the machine and an error if any occurred during the retrieval process.
 func (r *MachineDeletionRemediationReconciler) getMachine(ctx context.Context, remediation *v1alpha1.MachineDeletionRemediation) (*machinev1beta1.Machine, error) {
-	// If the Machine's Name and Ns come from the related Node, it is expected
-	// to find the Machine in the cluster, while if its Name and Ns come from
-	// CR's annotation, the Machine might have been deleted upon our request.
+	// If the remediation does not have the MachineNameNsAnnotation, we will try to get the Machine's data from one of
+	// the following sources:
+	// - the remediation's ownerReference, if the remediation was created by MachineHealthcheck
+	// - the remediation's Node, if the remediation was created by NodeHealthcheck
+
 	machineName, machineNs, err := getRemediationDataFromAnnotation(remediation, MachineNameNsAnnotation)
 	if err != nil {
 		r.Log.Error(err, "could not get Machine data from remediation", "remediation", remediation.GetName(), "annotation", MachineNameNsAnnotation)
 		return nil, unrecoverableError
 	}
 
 	var mustExist bool
+
+	if machineName == "" && len(remediation.GetOwnerReferences()) > 0 {
+		mustExist = true
+
+		machineKind, machineNamespace, err := r.getMachineKindAndNamespace(ctx)
+		if err != nil {
+			return nil, err
+		}
+
+		for _, owner := range remediation.GetOwnerReferences() {
+			if owner.Kind == machineKind {
+				machineName, machineNs = owner.Name, machineNamespace
+				mustExist = true
```

Comment:
Is it really needed? At line 299 you already set it to true, regardless if owner.Kind == machineKind. I think one of this assigments is redundant

Reply:
You're right, the first one is reduntant, I've fixed it already in a local version